clang/lib/StaticAnalyzer/Checkers/WebKit/ASTUtils.cpp

```diff
@@ -163,10 +163,13 @@ bool isConstOwnerPtrMemberExpr(const clang::Expr *E) {
     if (OCE->getOperator() == OO_Star && OCE->getNumArgs() == 1)
       E = OCE->getArg(0);
   }
-  auto *ME = dyn_cast<MemberExpr>(E);
-  if (!ME)
+  const ValueDecl *D = nullptr;
+  if (auto *ME = dyn_cast<MemberExpr>(E))
+    D = ME->getMemberDecl();
+  else if (auto *IVR = dyn_cast<ObjCIvarRefExpr>(E))
+    D = IVR->getDecl();
+  else
```

Comment:
This `else` clause already returns if D is a nullptr. Do we still need the check at line 173 below?

Reply:
Oh, that's a good point. We don't need this else clause since we're checking !D in line 173. Will remove.